### PR TITLE
Add installation instructions to the README

### PR DESCRIPTION
**Pull request by nilslindemann**
Feel free to modify as you desire, @paldepind

### CODE REVIEW

**Comment by iambumblehead** [comment]
the browser test pipeline fails intermittently and frequently on first-push

>  ❌ The browser was unable to create and start a test page after 30000ms. You can increase this timeout with the browserStartTimeout option.  (failed on firefox)

if you push up any sort of small change to trigger the pipeline it usually passes on the second try

**Comment by nilslindemann** [comment]
@paldepind 

> rename section to "How to run the examples"

That is a good point. See my commit "[Rename and move installing instructions](https://github.com/snabbdom/snabbdom/pull/1088/commits/7214e0951c0dfc7060de081f010e6bf6e635e8e0)". In that new context, I also moved the instructions below the section "More examples". Let me know if there are issues.

> Add "installation" section to CONTRIBUTING.md which also explains what users have to do to install a library in their own project, and for people who want to contribute.

That is also an excellent idea. I do not know how to install snabbdom as a library, so I would like to ask someone more experienced than me to do that :-)

**Comment by nilslindemann** [comment]
Sorry, I saw too late that you use a [specific commit format](https://www.conventionalcommits.org/en/v1.0.0/).

File: README.md [hunk 1]
```diff
@@ -118,16 +118,35 @@ patch(vnode, newVnode); // Snabbdom efficiently updates the old view to the new
...
+To run above examples locally:
+
+* As usual, Node.js and git needs to be installed on your system.
+* Clone/download [the snabbdom repository](https://github.com/snabbdom/snabbdom).
+* In the repository root (where the `package.json` is):
+  * Install the requirements: `npm install`.
+  * Run the tests (this also builds `/examples`):  `npm test`. Alternatively, just build the examples: `npm run build`.
+  * Start a local server, for example by running: `npx serve`. Open [localhost:3000](http://localhost:3000/).
+    This will redirect to [localhost:3000/examples](http://localhost:3000/examples) (due to the `./serve.json`).
+    Inspect the examples.
```
Comment (kuraga) [inline README.md:132]: @nilslindemann , thanks!

Let's simplify? :)

```suggestion
### How to run the examples and the tests locally

To run above examples and the tests locally, as usual, Node.js, NPM and git needs to be installed on your system.
```bash
git clone https://github.com/snabbdom/snabbdom  # clone the snabbdom repository
cd snabbdom
npm install  # install the requirements
```

To inspect the examples:
```bash
npm run build  # build the examples
npx serve  # start a local server
# open http://localhost:3000/examples in your browser
```
To run the tests:
```bash
npm test
```
```

Comment (nilslindemann) [reply README.md]: Thanks @kuraga, I have applied your suggestions in the latest commit

**Comment by nilslindemann** [comment]
I would be very interested, as @paldepind already mentioned, to also add a paragraph, how to add snabbdom as a project dependency. I just tried to do that by adding it to the dev dependencies in `package.json`, but failed. Could someone help out here?

Also, I think we could then move that section as simply "Installation" below the TOC.

**Comment by kuraga** [comment]
@nilslindemann , `npm install snabbdom` inside the project's root. Or, add `snabbdom` in `package.json`, then `npm install`.

File: README.md [hunk 2]
```diff
@@ -118,16 +118,48 @@ patch(vnode, newVnode); // Snabbdom efficiently updates the old view to the new
...
+npm run build  # build the examples
+npx serve  # start a local server
+# open http://localhost:3000/examples in your browser
+```
+
+To run the tests:
+
+```bash
+npm test
+````
```
Comment (kuraga) [inline README.md:145]: There is one redundant tilde.

File: README.md [hunk 3]
```diff
@@ -118,16 +118,48 @@ patch(vnode, newVnode); // Snabbdom efficiently updates the old view to the new
...
+### How to run the examples and tests locally
+
+As usual, Node.js, NPM and git are needed to be installed on your system.
+
+To install snabbdom locally:
+
+```bash
+git clone https://github.com/snabbdom/snabbdom  # clone the snabbdom repository
+cd snabbdom
+npm install  # install the requirements
```
Comment (kuraga) [inline README.md:130]:
```suggestion
npm install  # install the dependencies
```

File: README.md [hunk 4]
```diff
@@ -118,16 +118,48 @@ patch(vnode, newVnode); // Snabbdom efficiently updates the old view to the new
 - [Hero transitions](http://snabbdom.github.io/snabbdom/examples/hero/)
 - [SVG Carousel](http://snabbdom.github.io/snabbdom/examples/carousel-svg/)
 
+### How to run the examples and tests locally
```
Comment (kuraga) [inline README.md:121]:
```suggestion
## How to run the examples and the tests
```

File: README.md [hunk 5]
```diff
@@ -118,16 +118,48 @@ patch(vnode, newVnode); // Snabbdom efficiently updates the old view to the new
...
+
 ---
 
 ## Table of contents
 
+- [Introduction](#introduction)
+- [Features](#features)
+- [Example](#example)
+- [More examples](#more-examples)
+  - [How to run the examples and tests locally](#how-to-run-the-examples-and-tests-locally)
```
Comment (kuraga) [inline README.md:155]: Move left?

**Comment by kuraga** [comment]
> Also, I think we could then move that section as simply "Installation" below the TOC (or into `CONTRIBUTING.md` and point to it, I do not mind where it is, as long as it is somewhere)

Add the `## Installation` section (one of the first ones).

**Comment by nilslindemann** [comment]
@kuraga

> `npm install snabbdom` inside the project's root. Or, add `snabbdom` in `package.json`, then `npm install`.

Ok, it is a few months since I did this last and I forgot a few things, it seems.

This is what I did:

Create a folder `snabbdom-test`

```
cd snabbdom-test
npm init
npm install snabbdom
```

Create a file `examples/index.html`<sup>1</sup> with the contents:

```
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <title>snabbdom Test</title>
</head>

<body>
    <div id="container"></div>
    <script src="examples/logic.js"></script>
</body>

</html>
```

Create a file `examples/logic.js` [1] having the [example given in the README](https://github.com/snabbdom/snabbdom#example) as content.

Run `npx serve`.

Now the browser console gives `Uncaught SyntaxError: Cannot use import statement outside a module (at logic.js:1:1)` This is the point where I forgot what I need to do. Can you give me a hint, @kuraga ?

<sup>1</sup> in a folder `examples`, because `npx serve` redirects to this folder. Dunno why it does that.

**Comment by kuraga** [comment]
@nilslindemann , well. "install to get the `snabbdom` module importable" (in the server script) and "install to get the Snabbdom working in the browser" are different intentions :)

Now, you're able to run something like [the Example](https://github.com/snabbdom/snabbdom#example) in `nodejs` REPL.

But yeah, a good point to mention.

To others: Rollup, Webpack?

**Comment by nilslindemann** [comment]
So, @kuraga, I would love to have this documented. We should do that.

* What could be a minimal node.js helloworld repl example which does something useful, but does it with snabbdom technology?
* Minimal browser example with webpack/Rollup?

I will for the moment just commit your above fixes but leave the chapter where it currently is.

**Pull request closed without merging**